keyboards/viterbi/keymaps/mike808/keymap.c

```diff
@@ -0,0 +1,158 @@
+  //|----+----+----+----+----+----+----|    |----+----+----+----+----+----+----|
+     G_X ,CAPS, A  ,O_C ,E_A ,U_G , I  ,      D  ,H_G ,T_A ,N_C , S  ,MINS, UP ,
+  //|----+----+----+----+----+----+----|    |----+----+----+----+----+----+----|
+     G_C ,LSFT,SCLN, Q  , J  , K  , X  ,      B  , M  , W  , V  , Z  ,RSFT,DOWN,
+  //|----+----+----+----+----+----+----|    |----+----+----+----+----+----+----|
+     G_V ,FN1 ,G_TB,G_SP,BSPC,SPC ,SPC ,     ENTM,ENTM,G_Z ,GRV ,BSLS,LEFT,RGHT
+  //`----+----+----+----+----+----+----'    `----+----+----+----+----+----+----'
+  ),
+
+  [_FN1] = LAYOUT_kc(
+  //,----+----+----+----+----+----+----.    ,----+----+----+----+----+----+----.
+         ,    , F1 , F2 , F3 , F4 , F5 ,      F6 , F7 , F8 , F9 ,F10 ,F11 ,    ,
+  //|----+----+----+----+----+----+----|    |----+----+----+----+----+----+----|
+         ,    ,EXLM, AT ,LCBR,RCBR,PIPE,         , P7 , P8 , P9 ,ASTR,F12 ,    ,
+  //|----+----+----+----+----+----+----|    |----+----+----+----+----+----+----|
+         ,    ,HASH,DLR ,LPRN,RPRN,GRV ,         , P4 , P5 , P6 ,PLUS,MINS,    ,
+  //|----+----+----+----+----+----+----|    |----+----+----+----+----+----+----|
+         ,    ,PERC,CIRC,LBRC,RBRC,TILD,     AMPR, P1 , P2 , P3 ,SLSH,    ,    ,
+  //|----+----+----+----+----+----+----|    |----+----+----+----+----+----+----|
+         ,    ,    ,    ,    ,    ,    ,         ,    ,PDOT, P0 ,EQL ,    ,    
+  //`----+----+----+----+----+----+----'    `----+----+----+----+----+----+----'
+  ),
+
+  [_MOUSE] = LAYOUT_kc(
+  //,----+----+----+----+----+----+----.    ,----+----+----+----+----+----+----.
+         ,    ,    ,    ,    ,    ,    ,         ,    ,    ,    ,    ,    ,    ,
+  //|----+----+----+----+----+----+----|    |----+----+----+----+----+----+----|
+         ,    ,    ,    ,MS_U,    ,    ,         ,    ,    ,    ,    ,    ,    ,
+  //|----+----+----+----+----+----+----|    |----+----+----+----+----+----+----|
+         ,    ,    ,MS_L,MS_D,MS_R,    ,         ,    ,    ,    ,    ,    ,    ,
+  //|----+----+----+----+----+----+----|    |----+----+----+----+----+----+----|
+         ,    ,    ,    ,    ,    ,    ,         ,    ,    ,    ,    ,    ,    ,
+  //|----+----+----+----+----+----+----|    |----+----+----+----+----+----+----|
+         ,    ,    ,    ,BTN1,BTN2,    ,         ,    ,    ,    ,    ,    ,    
+  //`----+----+----+----+----+----+----'    `----+----+----+----+----+----+----'
+  )
+};
+
+#ifdef AUDIO_ENABLE
+float tone_qwerty[][2]     = SONG(QWERTY_SOUND);
+float tone_dvorak[][2]     = SONG(DVORAK_SOUND);
+#endif
+
+void update_rgblight(uint16_t layer) {
+  if (layer & (1UL << _DVORAK)) {
+    rgblight_sethsv_green();
+  } else if (layer & (1UL << _QWERTY)) {
+    rgblight_sethsv_goldenrod();
+  }
+}
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+  #ifdef RGBLIGHT_ENABLE
+  update_rgblight(default_layer);
```

Comment:
Purely optional/cosmetic...


`layer_state_set_user` could still be used here, actually. 
In fact, this should work: 

```
uint32_t layer_state_set_user(uint32_t state) {
#ifdef RGBLIGHT_ENABLE
  uint8_t default_layer = eeconfig_read_default_layer();
  switch (biton32(state)) {
    case _QWERTY:
       if (default_layer & (1UL << _DVORAK)) {
        rgblight_sethsv_green();
      }
      else {
        rgblight_sethsv_goldenrod();
      }
      break;
  }
#endif // RGBLIGHT_ENABLE
  return state;
}
```
Which would mean that you could use `set_single_persistent_default_layer(_QWERTY);`, rather than `persistent_default_layer_set`.